src/solver/simulation/include/antares/solver/simulation/remix-storage/storage-for-remix-no-levels.h

```diff
@@ -6,23 +6,27 @@
 
 namespace Antares::Solver::Simulation
 {
-
 constexpr double TOLERANCE = 1.e-6;
-const std::string error_msg_start_hydro_remix = "Remix hydro input : ";
 
 class StorageForRemixNoLevels: public IStorageForRemix
 {
 public:
     StorageForRemixNoLevels(std::vector<double>& generation,
                             std::vector<double>& unsupE,
                             const std::vector<double> Pmax,
-                            const std::vector<double> Pmin);
+                            const std::vector<double> Pmin,
+                            const std::string& name = "");
 
     double maxExchange(unsigned hourOfMaxGen, unsigned hourOfMinGen) const override;
     void update() override;
     const std::vector<double>& initWithdrawal() const override;
     std::vector<double>& withdrawal() override;
 
+    const std::string& name() const override
```

Comment:
Any storage for remix has name, we know that.
Indeed, interface class **IStorageForRemix** has a method **name()**.
So why not have the interface class hold the data member **name_** and have the `const std::string& name()` method ?
A (small) benefit will be that when calling name(), we won't have the (small) overhead due to polymorphism.